backend/vapid.js

```diff
@@ -0,0 +1,95 @@
+import sbp from '@sbp/sbp'
+
+let vapidPublicKey: string
+let vapidPrivateKey: Object
+
+// The Voluntary Application Server Identification (VAPID) email field is "a
+// stable identity for the application server" that "can be used by a push
+// service to establish behavioral expectations for an application server"
+// RFC 8292
+if (!process.env.VAPID_EMAIL) {
+  console.warn('Missing VAPID identification. Please set VAPID_EMAIL to a value like "mailto:some@example".')
+}
+const vapid = { VAPID_EMAIL: process.env.VAPID_EMAIL || 'mailto:test@example.com' }
+
+export const initVapid = async () => {
+  const vapidKeyPair = await sbp('chelonia/db/get', '_private_immutable_vapid_key').then(async (vapidKeyPair: string): Promise<[Object, string]> => {
+    if (!vapidKeyPair) {
+      // Generate a new ECDSA key pair
+      const keyPair = await crypto.subtle.generateKey(
+        {
+          name: 'ECDSA',
+          namedCurve: 'P-256' // Use P-256 curve
+        },
+        true, // Whether the key is extractable
+        ['sign', 'verify'] // Usages
+      )
+
+      // Export the private key
+      const serializedKeyPair = await Promise.all([
+        crypto.subtle.exportKey('jwk', keyPair.privateKey),
+        crypto.subtle.exportKey('raw', keyPair.publicKey).then((key) =>
+          // $FlowFixMe[incompatible-call]
+          Buffer.from(key).toString('base64url')
+        )
+      ])
+
+      return sbp('chelonia/db/set', '_private_immutable_vapid_key', JSON.stringify(serializedKeyPair)).then(() => {
+        return [keyPair.privateKey, serializedKeyPair[1]]
+      })
+    }
+
+    const serializedKeyPair = JSON.parse(vapidKeyPair)
+    return [
+      await crypto.subtle.importKey(
+        'jwk',
+        serializedKeyPair[0],
+        { name: 'ECDSA', namedCurve: 'P-256' },
+        false,
+        ['sign']
+      ),
+      serializedKeyPair[1]
+    ]
+  })
+
+  vapidPrivateKey = vapidKeyPair[0]
+  vapidPublicKey = vapidKeyPair[1]
+}
+
+const generateJwt = async (endpoint: URL): Promise<string> => {
```

Comment:
```js
  if (!vapidPrivateKey) {
    throw new Error('VAPID keys not initialized. Call initVapid first.')
  }
```

Reply:
This situation isn't really possible the way things are set up. This is only called when sending push notifications, which is only done after calling `initVapid`.

Reply:
For reference, upon your suggestion, we already do this for other things that need to be initialised. Although the check is technically correct, I'd rather not add it because it's unnecessary (and if it's necessary, it's a bug that needs to be fixed).